### PR TITLE
Improve Coverage and Instructiveness for branchTest.js

### PR DESCRIPTION
**Pull request by beaushinkle**
references https://github.com/keep-network/sortition-pools/issues/161

This branch tries to improve `branchTest.js` in two main ways:

1) We attempt to be more clear about what we're doing through the use of comments and more explicit math

2) We provide many more test cases in the hopes of building intuition (and increasing code coverage).

This PR lacks any additional tests for failure cases or edge cases, but I consider that a stretch goal and want to work on shoring up the current tests first.

tagging @eth-r and @nkuba for review.

### CODE REVIEW

**Review by eth-r — DISMISSED** [review]

File: test/branchTest.js [hunk 1]
```diff
@@ -20,37 +25,82 @@ describe("Branch", () => {
...
     it("Returns the uint16 in the correct position", async () => {
-      const result = await branchInstance.getSlot(node, 3)
-      expect(result).to.be.equal(0x33333333)
+      const expectations = [
+        0x00000000, 0x11111111, 0x22222222, 0x33333333, 0x44444444, 0x55555555,
+        0x66666666, 0x77777777,
+      ]
+      for (let slot = 0; slot < 8; slot++) {
+        const result = await branchInstance.getSlot(node, slot)
+        expect(result).to.equal(expectations[slot])
```
Comment (lukasz-zimnoch) [inline test/branchTest.js:34]: Worth adding an error message that will indicate the problematic slot:
```suggestion
        expect(result).to.equal(expectations[slot], `unexpected result at position ${slot}`)
```

**Comment by beaushinkle** [comment]
Sweet - I added the suggested changes, and then tampered with the test data to see some outputs:
```
      unexpected result at position 1
      + expected - actual

      -0x7777777766666666555555554444444433333333222222220000000000000000
      +0x7777777766666666555555554444444433333333222222220000001000000000
      ```
      
      LGTM

File: test/branchTest.js (hunk 1 above)
Comment (beaushinkle) [reply test/branchTest.js]: [b665a29](https://github.com/keep-network/sortition-pools/pull/172/commits/b665a29858ed4d95052032a36ea08f143676740a)

File: test/branchTest.js [hunk 2]
```diff
@@ -20,37 +25,82 @@ describe("Branch", () => {
...
+        "0x7777777766666666000000004444444433333333222222221111111100000000",
+        "0x7777777700000000555555554444444433333333222222221111111100000000",
+        "0x66666666555555554444444433333333222222221111111100000000",
+      ]
 
-      const result = await branchInstance.clearSlot(node, 3)
-      expect(ethers.utils.hexlify(result)).to.be.equal(newNode)
+      for (let slot = 0; slot < 8; slot++) {
+        const result = await branchInstance.clearSlot(node, slot)
+        expect(ethers.utils.hexlify(result)).to.equal(expectations[slot])
```
[the remark this answers is not in the document]
Comment (beaushinkle) [reply test/branchTest.js]: [b665a29](https://github.com/keep-network/sortition-pools/pull/172/commits/b665a29858ed4d95052032a36ea08f143676740a)

File: test/branchTest.js [hunk 3]
```diff
@@ -20,37 +25,82 @@ describe("Branch", () => {
...
+        "0x7777777766666666555555554444444433333333123456781111111100000000",
+        "0x7777777766666666555555554444444412345678222222221111111100000000",
+        "0x7777777766666666555555551234567833333333222222221111111100000000",
+        "0x7777777766666666123456784444444433333333222222221111111100000000",
+        "0x7777777712345678555555554444444433333333222222221111111100000000",
+        "0x1234567866666666555555554444444433333333222222221111111100000000",
+      ]
+      for (let slot = 0; slot < 8; slot++) {
+        const result = await branchInstance.setSlot(node, slot, newWeight)
+        expect(ethers.utils.hexlify(result)).to.equal(expectations[slot])
```
[the remark this answers is not in the document]
Comment (beaushinkle) [reply test/branchTest.js]: [b665a29](https://github.com/keep-network/sortition-pools/pull/172/commits/b665a29858ed4d95052032a36ea08f143676740a)

File: test/branchTest.js [hunk 4]
```diff
@@ -20,37 +25,82 @@ describe("Branch", () => {
...
 
-      const modified = await branchInstance.setSlot(node, 3, w)
-      newSlot = await branchInstance.getSlot(modified, 3)
-      expect(ethers.utils.hexlify(modified)).to.be.equal(newNode)
+        const newSlot = await branchInstance.getSlot(result, slot)
+        expect(newSlot).to.equal(newWeight)
```
[the remark this answers is not in the document]
Comment (beaushinkle) [reply test/branchTest.js]: [b665a29](https://github.com/keep-network/sortition-pools/pull/172/commits/b665a29858ed4d95052032a36ea08f143676740a)

**Review by lukasz-zimnoch — APPROVED** [review]
LGTM!
